erpnext/hr/doctype/employee_attendance_log/employee_attendance_log.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.utils import now
+from frappe.model.document import Document
+from frappe import _
+
+class EmployeeAttendanceLog(Document):
+	def validate(self):
+		if frappe.db.exists('Employee Attendance Log', {'employee': self.employee, 'time': self.time}):
+			frappe.throw('This log already exists for this employee.')
+
+
+@frappe.whitelist()
+def add_log_based_on_biometric_rf_id(biometric_rf_id, timestamp, device_id=None, log_type=None):
```

Comment:
fieldname should be `attendance_device_id` not `biometric_rf_id`

It may not be biometric, could be a card scan too!

Reply:
Yes, It was named `'Biometric/RF ID'` in the UI. I have now changed it to `attendance_device_id`